### PR TITLE
Add OKE oracle_cloud container_manager

### PR DESCRIPTION
**Pull request by agrare**
Screenshot from 2021-08-25 10-23-54
Screenshot from 2021-08-25 10-24-50

TODO:
- [x] DDF params
- [x] VCR specs
- [x] Test streaming watch_thread (default token valid for 4mins)

Depends on:
- [x] https://github.com/ManageIQ/manageiq-providers-kubernetes/pull/444

Follow-up:
- [x] https://github.com/ManageIQ/manageiq-providers-kubernetes/issues/446
- [x] https://github.com/ManageIQ/manageiq-decorators/pull/53

https://github.com/ManageIQ/manageiq-providers-oracle_cloud/issues/31

### CODE REVIEW

File: app/models/manageiq/providers/oracle_cloud/container_manager.rb [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      signer.sign(:GET, url, params, nil)
+
+      url.query = params.to_query
+
+      Base64.urlsafe_encode64(url.to_s)
+    end
+
+    private
+
+    def oci_config(tenant, user, private_key, public_key, region)
```
Comment (agrare) [inline app/models/manageiq/providers/oracle_cloud/container_manager.rb:96]: TODO this is the same code as `ManagerMixin.raw_connect` and should be refactored

File: app/models/manageiq/providers/oracle_cloud/container_manager/refresh_worker/watch_thread.rb [hunk 2]
```diff
@@ -1,2 +1,38 @@
 class ManageIQ::Providers::OracleCloud::ContainerManager::RefreshWorker::WatchThread < ManageIQ::Providers::Kubernetes::ContainerManager::RefreshWorker::WatchThread
+  def self.start!(ems, queue, entity_type, resource_version)
+    auth = ems.authentication_tokens.first
+
+    # Since we have to reconnect with a new bearer token we can't simply save the
+    # connect_options for reconnecting.  We have to save the info that we need
+    # to request a new bearer token from Oracle Cloud.
+    connect_options = {
```
[the remark this answers is not in the document]
Comment (agrare) [reply app/models/manageiq/providers/oracle_cloud/container_manager/refresh_worker/watch_thread.rb]: Renamed to oci_connect_options :shrug: not sure that's better it is just a local var and being assigned to `WatchThread.@connect_options` but it is different

**Comment by agrare** [comment]
@Fryguy PTAL

File: app/models/manageiq/providers/oracle_cloud/container_manager/refresh_worker/watch_thread.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (agrare) [reply app/models/manageiq/providers/oracle_cloud/container_manager/refresh_worker/watch_thread.rb]: Okay, worked out a much better way of doing this.  The logic to re-set the bearer token is contained in the `OracleCloud::ContainerManager.kubernetes_auth_options` at the class level so we don't have to change anything in the watch_thread

**Comment by Fryguy** [comment]
Backported to `morphy` in commit 1e62bccb5876e9ce3a744160180afc36b90caf03.

```text
commit 1e62bccb5876e9ce3a744160180afc36b90caf03
Author: Jason Frey <fryguy9@gmail.com>
Date:   Wed Sep 1 15:31:29 2021 -0400

    Merge pull request #30 from agrare/add_oke_container_manager
    
    Add OKE oracle_cloud container_manager
    
    (cherry picked from commit 8b239260a643c71b7645eabb4493217e19901290)
```

**Pull request merged**